### ISSUE TITLE
[Question] - Why have the --cap-add sys_resource and --privileged to run the cluster locally?

### ISSUE THREAD

**Issue by jay-johnson** [RedisLabs/DockerHub#1]
First I want to say thanks for releasing this. I had a couple questions around why the cap add and privileged is required to run the redis docker cluster?

``docker exec -d --privileged $rp_container_name_prefix "/opt/redislabs/bin/rladmin" cluster create name $rp_fqdn username $rp_admin_account_name password $rp_admin_account_password flash_enabled``

``docker run -d --cap-add sys_resource --name rp -p 8443:8443 -p 12000:12000 redislabs/redis``

1. Examining the test code:

https://github.com/RedisLabs/DockerHub/blob/master/utils/test_repo.sh#L42
https://github.com/RedisLabs/DockerHub/blob/master/utils/test_repo.sh#L76

2. Docker details:
https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities

``When the operator executes docker run --privileged, Docker will enable to access to all devices on the host as well as set some configuration in AppArmor or SELinux to allow the container nearly all the same access to the host as processes running outside containers on the host. Additional information about running with --privileged is available on the Docker Blog.``

``SYS_RESOURCE - Override resource Limits.``

3. Are there specific resources the cluster needs that can be used with a combination of these constraints that make it clear what resources are recommended/required?
https://docs.docker.com/engine/reference/run/#runtime-constraints-on-resources

**Comment by cihanb** [comment]
thanks @jay-johnson, 
For the quick and dirty test code, I added the code to sanity check the versions on all OS flavors (win, mac,linux). I believe I added this because, I hit a problem when running the repo_test under git-bash on windows. I worked around it with --privileged with "docker exec". Let me debug one more time to see if I can get docker exec without privileged execution working.

regarding --cap-add sys_resource is needed for all the privileges needed for the container. Flag is needed for capabilities on linux for the clustering capabilities in redis-enterprise. During the beta phase of the image, we plan to go through the process of making the image "official" on the hub and we'll review it with docker folks to see if there are better ways for our engine run with less privileges but for now during the beta period we require the --cap-add. @ushachar may have additional comments.
